File: EasyFinance.Domain/Financial/ExpenseItem.cs
using System;
using System.Collections.Generic;
using EasyFinance.Domain.AccessControl;

namespace EasyFinance.Domain.Financial
{
    public class ExpenseItem : BaseExpense
    {
        private ExpenseItem() { }

        public ExpenseItem(
            string name = "default",
            DateOnly date = default,
            decimal amount = default,
            User createdBy = default,
            ICollection<Attachment> attachments = default,
            ICollection<ExpenseItem> items = default)
            : base(name, date, amount, createdBy, attachments, items)
        {
        }
    }
}

File: easyfinance.client/src/app/core/models/expense-item.ts
import { AutoMap } from "@automapper/classes";

export class ExpenseItem {
  @AutoMap()
  id!: string;
  @AutoMap()
  name!: string;
  @AutoMap()
  date!: Date;
  @AutoMap()
  amount!: number;
  @AutoMap(() => [ExpenseItem])
  items!: ExpenseItem[];
}
